Add pull request templates

MarvinLira opened the pull request:
<!--
  Have you read our Code of Conduct?
  By filing an issue or pull request, you are expected to comply with it, including treating everyone with respect:
  https://github.com/FreeProving/guidelines/blob/master/CODE_OF_CONDUCT.md
-->

### Issue

We currently already have templates for issues but we have no templates for PRs yet.

### Description of the Change

I have added a template for PRs based on the templates in the  [repository of Atom](https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE).

### Alternate Designs

I started by creating specific templates for PRs as Atom also uses multiple templates.
But those templates didn't differ much and I decided to merge them into a single template.

### Possible Drawbacks

### Verification Process

MarvinLira committed:
Update readme

MarvinLira committed:
Add two templates for PRs

MarvinLira committed:
Add generic PR template

MarvinLira committed:
Remove first two PR templates

Both differ not much from the generic template.

just95 commented on PULL_REQUEST_TEMPLATE/generic.md:
```diff
@@ -0,0 +1,43 @@
+<!--
+  Have you read our Code of Conduct?
+  By filing an issue or pull request, you are expected to comply with it, including treating everyone with respect:
+  https://github.com/FreeProving/guidelines/blob/master/CODE_OF_CONDUCT.md
```
The default branch has recently been renamed from `master` to `main` (see #2).

```suggestion
  https://github.com/FreeProving/guidelines/blob/main/CODE_OF_CONDUCT.md
```

just95 commented on PULL_REQUEST_TEMPLATE/generic.md:
```diff
@@ -0,0 +1,43 @@
+
+### Alternate Designs
+
+<!-- Explain what other alternates were considered and why the proposed version was selected -->
```
```suggestion
<!-- Explain what other alternates were considered and why the proposed version was selected. -->
```

just95 commented on PULL_REQUEST_TEMPLATE/generic.md:
```diff
@@ -0,0 +1,43 @@
+
+Give any additional information that is not covered by the linked issue but might be important for the reviewer.
+
+-->
```
The GitHub issue and pull request comment editor does not provide any syntax highlighting. Thus, it is difficult to tell where Markdown comments start and where they end. Therefore, I would avoid blank lines in Markdown comments such that the user does not insert the text between `<!-- … -->` by accident. Furthermore, I used some indentation in the issue templates to make start and end of the comment stand out more.

```suggestion
<!--
  Link to the issue that this pull request addresses.
  If there is not yet a corresponding issue, please open a new issue and then link to that issue in your pull request.
  Give any additional information that is not covered by the linked issue but might be important for the reviewer.
-->
```

just95 commented:
Oh, and we should maybe also mention the template in the *Create a Pull Request* section of the [CONTRIBUTING.md](https://github.com/FreeProving/guidelines/blob/main/CONTRIBUTING.md#contributing-code).

MarvinLira committed:
Apply suggestions

- Replace `.../master/...` by `.../main/...`.
- Add missing dot.
- Avoid blank lines in comments.
- Use indentation on comments.

MarvinLira committed:
Mention PR template in `CONTRIBUTING.md`
